src/api/ActionMapper.ts

```diff
@@ -26,9 +26,10 @@ export type Callbacks<T extends object> = {
   apply?: MappedRules
 };
 
-type ActionCreator<TargetFunction extends (...args: any[]) => any, Action> = TargetFunction extends Listeners<any>
-  ? (ev: EventOfListeners<TargetFunction>) => Action
-  : (...args: Parameters<TargetFunction>) => Action;
+type ActionCreator<TargetFunction, Action>
```

Comment:
Why was this necessary? Would you like to mention this in the commit message?

Reply:
Honestly not sure, TS4.8 just doesn't seem to like narrow type parameters on mapped types, which where this is used.